nixos/modules/programs/x2goserver.nix

```diff
@@ -0,0 +1,134 @@
+  x2goServerConf = pkgs.writeText "x2goserver.conf" confText;
+
+  x2goAgentOptions = pkgs.writeText "x2goagent.options" ''
+    X2GO_NXOPTIONS=""
+    X2GO_NXAGENT_DEFAULT_OPTIONS="${concatStringsSep " " cfg.nxagentDefaultOptions}"
+  '';
+
+in {
+  options.programs.x2goserver = {
+    enable = mkEnableOption "x2goserver";
+
+    superenicer = {
+      enable = mkEnableOption "superenicer" // {
+        description = ''
+          Enables the SupeReNicer code in x2gocleansessions, this will renice
+          suspended sessions to nice level 19 and renice them to level 0 if the
+          session becomes marked as running again
+        '';
+      };
+    };
+
+    nxagentDefaultOptions = mkOption {
+      type = types.listOf types.str;
+      default = [ "-extension GLX" "-nolisten tcp" ];
+      example = [ "-extension GLX" "-nolisten tcp" ];
+      description = ''
+        List of default nx agent options.
+      '';
+    };
+
+    settings = mkOption {
+      type = types.attrsOf types.attrs;
+      default = {};
+      description = ''
+        x2goserver.conf ini configuration as nix attributes. See
+        `x2goserver.conf(5)` for details
+      '';
+      example = literalExample ''
+        superenicer = {
+          "enable" = "yes";
+          "idle-nice-level" = 19;
+        };
+        telekinesis = { "enable" = "no"; };
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ pkgs.x2goserver ];
+
+    users.groups.x2go = {};
+    users.users.x2go = {
+      home = "/var/lib/x2go/db";
+      group = "x2go";
+    };
+
+    security.wrappers.x2gosqliteWrapper = {
+      source = "${pkgs.x2goserver}/lib/x2go/libx2go-server-db-sqlite3-wrapper.pl";
+      owner = "x2go";
+      group = "x2go";
+      setgid = true;
+    };
+    security.wrappers.x2goprintWrapper = {
+      source = "${pkgs.x2goserver}/bin/x2goprint";
+      owner = "x2go";
+      group = "x2go";
+      setgid = true;
+    };
+    # x2goclient sends SSH commands with preset PATH set to
+    # "/usr/local/bin;/usr/bin;/bin". Since we cannot filter arbitrary ssh
+    # commands, we have to make the following executables available.
+    system.activationScripts.x2goserver = with pkgs; ''
+      mkdir -p /usr/local/bin
+      for exe in ${x2goserver}/bin/x2go*
+      do
+        ln -sf $exe /usr/local/bin/
+      done
+      for util in \
+        ${xorg.xrandr}/bin/xrandr ${xorg.xmodmap}/bin/xmodmap \
+        ${utillinux}/bin/setsid ${gawk}/bin/awk ${gnused}/bin/sed \
+        ${coreutils}/bin/cp ${coreutils}/bin/chmod
+      do
+        ln -sf $util /usr/local/bin/
+      done
+      mkdir -p /var/lib/x2go/conf
+      cp -r ${x2goserver}/etc/x2go/* /var/lib/x2go/conf/
+      ln -sf ${x2goServerConf} /var/lib/x2go/conf/x2goserver.conf
+      ln -sf ${x2goAgentOptions} /var/lib/x2go/conf/x2goagent.options
+      if [ ! -f /var/lib/x2go/db/x2go_sessions ]
+      then
+        ${x2goserver}/bin/x2godbadmin --createdb
```

Comment:
Putting all those commands outside of an activationScript would make this PR a lot nicer imo. The ln/cp setup can be done with the `L+`/`C` rule of `systemd.tmpfiles` I think, and the `createdb` can be done in a prestart script of the systemd unit.

Reply:
That worked. It is nicer, although the list of binaries is a bit annoying, because I didn't find a way to iterate on `${x2goserver}/bin/*`.